File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = [[0]*n for _ in range(n)]
    answer2 = []
    cnt = 0
    for i, j in zip(arr1, arr2):
        l1 = []
        l2 = []
        while True:
            if i == 0:
                l1.append(0)
                break
            if i == 1:
                l1.append(1)
                break
            if i % 2 == 1:
                i = i//2
                l1.append(1)
            else:
                i = i//2
                l1.append(0)
                
        lenl = len(l1)
        for _ in range(lenl,n):
            l1.append(0)
            
        while True:
            if j == 0:
                l2.append(0)
                break
            if j == 1:
                l2.append(1)
                break
            if j % 2 == 1:
                j = j//2
                l2.append(1)
            else:
                j = j//2
                l2.append(0)

        lenl = len(l2)
        for _ in range(lenl,n):
            l2.append(0)
            
        for b in range(n-1, -1, -1):
                if (l1[n-b-1] == 1) or (l2[n-b-1] == 1):
                    answer[cnt][b] = '#'
                else:
                    answer[cnt][b] = ' '
        cnt += 1
    
    for i in range(n):
        tmp = ""
        for j in range(n):
            if answer[i][j] == '#':
                tmp += "#"
            else:
                tmp += " "
        answer2.append(tmp)
    return answer2

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
import java.util.*;

class Solution {
    char[][] tmpAnswer;
    String[] strArr1;
    String[] strArr2;
    
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer;
        
        tmpAnswer = new char[n][n];
        strArr1 = new String[n];
        strArr2 = new String[n];
        
        for (int i = 0; i < n; i++) {
            String tmp1 = Integer.toBinaryString(arr1[i]);
            
            if (tmp1.length() < n) {
                tmp1 = "0".repeat(n-tmp1.length()) + tmp1;
            }
            
            String tmp2 = Integer.toBinaryString(arr2[i]);
            
            if (tmp2.length() < n) {
                tmp2 = "0".repeat(n-tmp2.length()) + tmp2;
            }
            
            for (int j = 0; j < n; j++) {
                if (tmp1.charAt(j) == '0' && tmp2.charAt(j) == '0') {
                    tmpAnswer[i][j] = ' ';
                    continue;
                }
                
                tmpAnswer[i][j] = '#';
            }
        }
        
        answer = new String[n];
        
        for (int i = 0; i < n; i++) {
            StringBuilder sb = new StringBuilder();
            
            for (int j = 0; j < n; j++) {
                sb.append(tmpAnswer[i][j]);
            }
            
            answer[i] = sb.toString();
        }
        
        return answer;
    }
}